pkg/logs/decoder/handler.go

```diff
@@ -19,6 +20,17 @@ type Handler interface {
 	Cleanup()
 }
 
+// NewLineHandler creates a new instance of general line handler based on the configuration.
+func NewLineHandler(source *config.LogSource, truncator LineTruncator) Handler {
+
+	for _, rule := range source.Config.ProcessingRules {
+		if rule.Type == config.MultiLine {
```

Comment:
Might worth adding a comment that we default to the first multiline configuration found.

Reply:
done